packages/remix/src/infrastructure/logto/handle-sign-up.ts

```diff
@@ -0,0 +1,55 @@
+import type { Session } from '@remix-run/node';
+
+import type { CreateLogtoClient } from './create-client.js';
+import type { LogtoStorage } from './create-storage.js';
+
+type HandleSignUpRequest = {
+  readonly redirectUri: string;
+};
+
+type HandleSignUpResponse = {
+  readonly session: Session;
+  readonly navigateToUrl: string;
+};
+
+class HandleSignUpCommand {
+  public static readonly fromDependencies = (dependencies: { createClient: CreateLogtoClient }) =>
+    new HandleSignUpCommand({ createClient: dependencies.createClient });
+
+  private navigateToUrl = '/api/sign-up';
+
+  private constructor(
+    private readonly properties: {
+      readonly createClient: CreateLogtoClient;
+    }
+  ) {}
+
+  public async execute(request: HandleSignUpRequest) {
+    const { createClient } = this.properties;
+
+    const client = createClient((url) => {
+      this.navigateToUrl = url;
+    });
+
+    await client.signIn(request.redirectUri, 'signUp');
```

Comment:
```suggestion
    await client.signIn({ redirectUri: request.redirectUri, interactionMode: 'signUp' });
```
The previous usage has been deprecated.

Reply:
fixed